plugins/airbrake/src/components/EntityAirbrakeWidget/EntityAirbrakeWidget.tsx

```diff
@@ -36,59 +37,70 @@ const useStyles = makeStyles<BackstageTheme>(() => ({
   },
 }));
 
+enum ComponentState {
+  Loading,
+  NoProjectId,
+  Error,
+  Loaded,
+}
+
 export const EntityAirbrakeWidget = ({ entity }: { entity: Entity }) => {
   const classes = useStyles();
 
   const projectId = useProjectId(entity);
-  const errorApi = useApi<ErrorApi>(errorApiRef);
   const airbrakeApi = useApi(airbrakeApiRef);
+  let componentState = ComponentState.Loading;
 
-  const { loading, value, error } = useAsync(
-    () => airbrakeApi.fetchGroups(projectId),
-    [airbrakeApi, projectId],
-  );
-
-  useEffect(() => {
-    if (error) {
-      errorApi.post(error);
-    }
-  }, [error, errorApi]);
-
-  if (loading || !projectId || error) {
-    return (
-      <InfoCard title="Airbrake groups" variant="gridItem">
-        {loading && <Progress />}
+  const { loading, value, error } = useAsync(() => {
+    return airbrakeApi.fetchGroups(projectId);
+  }, [airbrakeApi, projectId]);
 
-        {!loading && !projectId && (
-          <MissingAnnotationEmptyState
-            annotation={AIRBRAKE_PROJECT_ID_ANNOTATION}
-          />
-        )}
+  if (loading) {
+    componentState = ComponentState.Loading;
```

Comment:
Why not just do `return <Progress />;` here? And same for the rest of the cases. This enum just makes things more complex and adds indirection, it could be removed.